Assets/Project/Scripts/Core/Services/GameReferences/IGameReferences.cs

```diff
@@ -0,0 +1,17 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Popeye.Core.Services.GameReferences
+{
+    public class IGameReferences : MonoBehaviour
+    {
+        protected Transform _playerTransform;
+
+        public Transform GetPlayer()
+        {
+            return _playerTransform;
+        }
+
+    }
```

Comment:
This should be an interface, with only the following:
public interface IGameReferences
{
      void SetPlayer(Transform playerTransform);
      Transform GetPlayer();
}

Then, the implementation has the member variable for the Transform reference, and it doesn't need to be a MonoBehaviour either.
public class GameReferences : IGameReferences
{
      private Transform _playerTransform;
      void SetPlayer(Transform playerTransform) {...}
      Transform GetPlayer(){...}
}